velox/functions/prestosql/types/BingTileType.h

```diff
@@ -20,6 +20,50 @@
 
 namespace facebook::velox {
 
+constexpr uint8_t BING_TILE_VERSION = 0;
+constexpr uint8_t BING_TILE_MAX_ZOOM_LEVEL = 23;
+constexpr uint8_t BING_TILE_ZOOM_BIT_WIDTH = 5;
+constexpr uint8_t BING_TILE_VERSION_OFFSET = 63 - BING_TILE_ZOOM_BIT_WIDTH;
+constexpr uint8_t BING_TILE_ZOOM_OFFSET = 31 - BING_TILE_ZOOM_BIT_WIDTH;
+constexpr uint64_t BITS_23_MASK = (1 << 24) - 1;
+constexpr uint64_t BITS_5_MASK = (1 << 6) - 1;
+
+inline uint64_t bingTileCoordsToInt(uint32_t x, uint32_t y, uint8_t zoom) {
+  uint64_t tile = (static_cast<uint64_t>(zoom) << BING_TILE_ZOOM_OFFSET) |
+      (static_cast<uint64_t>(x) << 32) | static_cast<uint64_t>(y);
+
+  return tile;
+}
+
+/// Returns the version of the tile (as uint64)
+inline uint8_t bingTileVersion(uint64_t tile) {
+  return (tile >> BING_TILE_VERSION_OFFSET) & BITS_5_MASK;
+}
+
+/// Returns the zoom of the tile (as uint64)
+inline uint8_t bingTileZoom(uint64_t tile) {
+  return (tile >> BING_TILE_ZOOM_OFFSET) & BITS_5_MASK;
+}
+
+/// Returns the x-coordinate of the tile (as uint64)
+inline uint8_t bingTileX(uint64_t tile) {
+  return (tile >> 32) & BITS_23_MASK;
+}
+
+/// Returns the y-coordinate of the tile (as uint64)
+inline uint8_t bingTileY(uint64_t tile) {
+  return tile & BITS_23_MASK;
+}
+
+/// Returns true if the tile (as uint64) is valid
+inline bool isBingTileIntValid(uint64_t tile) {
+  uint64_t coordinateBound = 1 << bingTileZoom(tile);
+  return (uint8_t)(bingTileVersion(tile) == BING_TILE_VERSION) &
```

Comment:
nit: just curious, is there any benefit to using `cast to (uint8_t) and using bitwise &` instead of using `booleans and logical &`

I would personally prefer the latter in favor of readability and avoiding of c-style casts.

Reply:
This way makes the operation branchless and only using SIMD-friendly instructions (bit shifts and comparisions).  Since we are doing this for all elements of a vector, it should make this very cheap: the cache lines will be prefetched and the instructions can be pipelined.

However there's a valid concern of premature optimization, so I'm happy to use standard short-circuiting `&&` and then only move to this if we have a benchmark that improves.

Reply:
Thanks for the explanation. I don't really have a strong preference either way so this works for me. Can you add a small comment explaining why we chose this, would be good for folks reading this later who are not familiar with this pattern.